src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -34,4 +34,15 @@ public BookDto findById(Long id) {
                 .orElseThrow(() -> new EntityNotFoundException("Can't get book by id" + id));
         return bookMapper.toDto(book);
     }
+    
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, BookRequestDto bookDto) {
+        Book book = bookMapper.toModel(bookDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
```

Comment:
we should check if a book by provided `id` even exists before doing some manipulations with it, we could add some private method for that